tests/README.md

```diff
@@ -0,0 +1,1510 @@
+which is living in the "\ILIAS\UI\Implementation\Component\Button\" namespace. In addition the only
+tested logic is the instantiation of the concrete button implementations.
+
+Furthermore this test checks 4 different things:
+- Standard button implements Standard interface ?
+- Primary button implements Primary interface ?
+- Close button implements Close interface ?
+- Shy button implements Shy interface ?
+
+```php
+<?php
+class ButtonTest extends ILIAS_UI_TestBase {
+
+	public function getButtonFactory() {
+		return new \ILIAS\UI\Implementation\Component\Button\Factory();
+	}
+
+	public function test_implements_factory_interface() {
+		$f = $this->getButtonFactory();
+
+		$this->assertInstanceOf("ILIAS\\UI\\Component\\Button\\Factory", $f);
+		$this->assertInstanceOf
+			( "ILIAS\\UI\\Component\\Button\\Standard"
+			, $f->standard("label", "http://www.ilias.de")
+			);
+		$this->assertInstanceOf
+			( "ILIAS\\UI\\Component\\Button\\Primary"
+			, $f->primary("label", "http://www.ilias.de")
+			);
+		$this->assertInstanceOf
+			( "ILIAS\\UI\\Component\\Button\\Close"
+			, $f->close()
+			);
+		$this->assertInstanceOf
+			( "ILIAS\\UI\\Component\\Button\\Shy"
+			, $f->shy("label", "http://www.ilias.de")
+			);
+	}
+}
+```
+
+<a name="example-solution"></a>
+##### Example solution
+The proposed solution would be to remove the test entirely because they test if the new keyword works. However, if the author wishes to keep the
+tests. They could be moved into the ButtonFactoryTest class and split up in smaller more precise tests.
+As a result the developer which runs the test is now able to see which part of the factory failed.
+
+```php
+<?php
+use \ILIAS\UI\Implementation\Component\Button\Factory;
+use \ILIAS\UI\Component\Button\Standard;
+use \ILIAS\UI\Component\Button\Primary;
+use \ILIAS\UI\Component\Button\Shy;
+use \ILIAS\UI\Component\Button\Close;
+
+class ButtonFactoryTest extends AbstractFactoryTest {
+
+	private $subject;
+
+	public function setUp() {
+		parent::setUp();
+
+		$this->subject = new Factory();
+	}
+
+	// ... other parts of the test class ...
+
+	/**
+	 * @test
+	 * @small
+	 */
+	public function test_creation_of_standard_button() {
+		$this->assertInstanceOf
+				( Standard::class
+				, $this->subject->standard("label", "http://www.ilias.de")
+				);
+	}
+
+	/**
+	 * @test
+	 * @small
+	 */
+	public function test_creation_of_primary_button() {
+		$this->assertInstanceOf
+					( Primary::class
+					, $this->subject->primary("label", "http://www.ilias.de")
+					);
+	}
+
+	/**
+	 * @test
+	 * @small
+	 */
+	public function test_creation_of_Shy_button() {
+		$this->assertInstanceOf
+					( Shy::class
+					, $this->subject->shy("label", "http://www.ilias.de")
+					);
+	}
+
+	/**
+	 * @test
+	 * @small
+	 */
+	public function test_creation_of_close_button() {
+		$this->assertInstanceOf
+					( Close::class
+					, $this->subject->close("label", "http://www.ilias.de")
+					);
+	}
+}
```

Comment:
Well you did not do much except splitting the test into multiple tests and giving the test itself a new name. Sure you can do that. However, there are far too many words and code somewhat covering those two rather simple acts. Please boil that down to the essence.

Reply:
It's only code, Timon -.o

Reply:
I think this rater simple act of splitting tests, is really important to improve the test feedback. For example the test `test_implements_factory_interface` can only express if something went wrong in the factory. However the tests in the example solution are providing a more precise feedback, we never have to ask "why" the test failed. Therefore, I would like to keep that example or make an abstract one because it highlights why it is useful to create small unit tests. Would it be ok if I shorten the example solution by two methods ?